core/src/main/java/amino/run/oms/migrationdecision/MetricWatcher.java

```diff
@@ -0,0 +1,282 @@
+                            @Override
+                            public int compare(
+                                    AbstractMap.SimpleEntry<ReplicaID, Long> replicaStat1,
+                                    AbstractMap.SimpleEntry<ReplicaID, Long> replicaStat2) {
+                                if (replicaStat1.getValue() < replicaStat2.getValue()) return 1;
+                                else if (replicaStat1.getValue() > replicaStat2.getValue())
+                                    return -1;
+                                return 0;
+                            }
+                        });
+        timer =
+                new ResettableTimer(
+                        new TimerTask() {
+                            public void run() {
+                                // get replica list
+                                List<ReplicaID> replicaIDs = getMigrationReplicas();
+                                if (replicaIDs.isEmpty()) {
+                                    logger.info("Replica not available for migration");
+                                    timer.reset();
+                                    return;
+                                }
+
+                                // get best candidates for migration
+                                List<ReplicaID> bestCandidateReplicaIds =
+                                        getBestCandidates(replicaIDs);
+                                logger.fine(
+                                        String.format(
+                                                "Chosen Replicas [%s]",
+                                                Arrays.toString(
+                                                        bestCandidateReplicaIds.toArray())));
+
+                                // try to migrate one of best replica candidate
+                                for (ReplicaID bestCandidateReplicaId : bestCandidateReplicaIds) {
+                                    // TODO check if recently migrated (check for cool down
+                                    //  time)
+                                    if (migrate(bestCandidateReplicaId)) {
+                                        timer.reset();
+                                        return;
+                                    }
+                                }
+                            }
+                        },
+                        METRIC_WATCH_INTERVAL);
+    }
+
+    /** Starts metric watcher */
+    public void start() {
+        timer.start();
+    }
+
+    private boolean migrate(ReplicaID replicaID) {
+        InetSocketAddress currentKernelServer;
+        try {
+            // get replica instance event handler
+            EventHandler replica = microServiceManager.getReplicaDispatcher(replicaID);
+            currentKernelServer = replica.getHost();
+        } catch (Exception e) {
+            logger.warning("Failed to get replica dispatcher");
+            e.printStackTrace();
+            return false;
```

Comment:
It seems it would be more appropriate to throw an exception here. 
What does the return value of this method actually mean?